build_tools/waf_dynamo.py

```diff
@@ -332,19 +332,23 @@ def default_flags(self):
         # Default to asmjs output
         wasm_enabled = 0
         legacy_vm_support = 1
-        binaryen_method = "asmjs"
         if 'wasm' == build_util.get_target_architecture():
             wasm_enabled = 1
             legacy_vm_support = 0
-            binaryen_method = "native-wasm"
+
+        emflags = ['WASM=%d' % wasm_enabled, 'LEGACY_VM_SUPPORT=%d' % legacy_vm_support, 'DISABLE_EXCEPTION_CATCHING=1', 'AGGRESSIVE_VARIABLE_ELIMINATION=1', 'PRECISE_F32=2',
+                   'EXTRA_EXPORTED_RUNTIME_METHODS=["stringToUTF8","ccall","stackTrace","UTF8ToString","callMain"]', 'EXPORTED_FUNCTIONS=["_main"]',
+                   'ERROR_ON_UNDEFINED_SYMBOLS=1', 'TOTAL_MEMORY=268435456']
+        emflags = zip(['-s'] * len(emflags), emflags)
+        emflags =[j for i in emflags for j in i]
```

Comment:
They don't clearly make a distinction between compile/link flags, and they're "safe to pass to both".

Reply:
This makes my head hurt "[j for i in emflags for j in i]" :-)